fauxfactory/__init__.py

```diff
@@ -673,33 +697,9 @@ def gen_utf8(length=10):
     # Validate length argument
     _is_positive_int(length)
 
-    # Generate codepoints. The valid range of UTF-8 codepoints is
-    # 0x0-0x10FFFF, minus the following: 0xC0-0xC1, 0xF5-0xFF and
-    # 0xD800-0xDFFF. These 2061 invalid codepoints (2 + 11 + 2048) comprise
-    # 0.2% of 0x0-0x10FFFF. Thus, it should be OK to just check for invalid
-    # codepoints and generate new ones if need be.
-    codepoints = []
-    while len(codepoints) < length:
-        # Use sys.maxunicode instead of 0x10FFFF to avoid the exception
-        # below, in a narrow Python build (before Python 3.3)
-        # ValueError: unichr() arg not in range(0x10000) (narrow Python
-        # build)
-        # For more information, read PEP 261.
-        codepoint = random.randint(0x0, sys.maxunicode)
-        if (
-                codepoint not in range(0xC0, 0xC1 + 1)
-                and codepoint not in range(0xF5, 0xFF + 1)
-                and codepoint not in range(0xD800, 0xDFFF + 1)):
-            codepoints.append(codepoint)
-
-    # Convert codepoints to characters. Python 2 and 3 support the `unichr`
-    # and `chr` functions, respectively.
-    if sys.version_info.major == 2:
-        # pylint:disable=E0602
-        output = u''.join(unichr(codepoint) for codepoint in codepoints)
-    else:
-        output = u''.join(chr(codepoint) for codepoint in codepoints)
-    return _make_unicode(output)
+    return _make_unicode(u''.join(
```

Comment:
Is the call to `_make_unicode` necessary here, given that the string being joined is `u''`?